rosbag2_transport/src/rosbag2_transport/recorder.cpp

```diff
@@ -160,8 +164,8 @@ RecorderImpl::RecorderImpl(
 : writer_(std::move(writer)),
   storage_options_(storage_options),
   record_options_(record_options),
-  stop_discovery_(record_options_.is_discovery_disabled),
   node(owner),
+  stop_discovery_(record_options_.is_discovery_disabled),
```

Comment:
this is required to avoid `recorder` warning since construction order has been changed.